lib/buttons-and-links/__tests__/KButton.spec.js

```diff
@@ -1,67 +1,109 @@
-    });
-    it('should render a dropdown icon when hasDropdown is true', () => {
-      const wrapper = shallowMount(KButton, {
-        propsData: {
-          hasDropdown: true,
-        },
+      it('should render a dropdown icon when hasDropdown is true', () => {
+        const wrapper = shallowMount(KButton, {
+          propsData: {
+            hasDropdown: true,
+          },
+        });
+        expect(wrapper.find('[data-test="dropdownIcon"]').exists()).toBe(true);
       });
-      expect(wrapper.find('[data-test="dropdownIcon"]').exists()).toBe(true);
     });
-  });
 
-  describe('text prop and slots', () => {
-    it('should render the text prop if nothing is in the default slot', () => {
-      const wrapper = shallowMount(KButton, {
-        propsData: {
-          text: 'test',
-        },
+    describe('text prop and slots', () => {
+      it('should render the text prop if nothing is in the default slot', () => {
+        const wrapper = shallowMount(KButton, {
+          propsData: {
+            text: 'test',
+          },
+        });
+        expect(wrapper.text()).toContain('test');
+      });
+
+      it('should render the slot when the slot has content', () => {
+        const wrapper = shallowMount(KButton, {
+          propsData: {
+            text: 'test',
+          },
+          slots: {
+            default: '<span>slot</span>',
+          },
+        });
+        expect(wrapper.text()).toContain('slot');
+        expect(wrapper.text()).toContain('test');
       });
-      expect(wrapper.text()).toContain('test');
     });
 
-    it('should render the slot when the slot has content', () => {
-      const wrapper = shallowMount(KButton, {
-        propsData: {
-          text: 'test',
-        },
-        slots: {
-          default: '<span>slot</span>',
-        },
+    describe('event handling', () => {
+      it('should emit a click event when clicked', () => {
+        const wrapper = shallowMount(KButton, {
+          propsData: {
+            text: 'test',
+          },
+        });
+        wrapper.trigger('click');
+        expect(wrapper.emitted().click).toBeTruthy();
       });
-      expect(wrapper.text()).toContain('slot');
-      expect(wrapper.text()).toContain('test');
     });
-  });
+  } else {
+    describe('KButton Visual Tests', () => {
+      beforeAll(async () => {
+        await page.goto('http://localhost:4000/testing-playground', { waitUntil: 'networkidle2' });
+      });
+
+      async function renderComponent(component, props) {
+        await page.evaluate(
+          ({ component, props }) => {
+            window.postMessage(
+              {
+                type: 'RENDER_COMPONENT',
+                component: component,
+                props: props,
+              },
+              '*'
+            );
+          },
+          { component, props }
+        );
+        await page.waitForSelector('#testing-playground');
```

Comment:
Something that could be wrong here is if we want to run several tests, when we take the second snapshot, we are not necessarily wating for it to re-render. Because the #testing-playground selector is always visible, and is not an indicator that the new render have already been made. I think we could use the `await this.$nextTick()` inside the component, but this would need direct method invocation or listeting to a response event. I think an alternative could be to define a method in the `window` object inside the testingPlayground component such as: 
``` javascript
mounted() {
  window.renderComponent = async (data) => {
        this.component = data.component;
        this.componentProps = data.props;
        await this.$nextTick();
  };
} 
```

And play with it to see how it goes. This way we can just call the await window.renderComponent(...); and when the promise resolves we will be ensuring that the re-render has already happened. 

If you have other Ideas how to ensure we are waiting for the re-render (or if the previous idea doesnt work) let us know and we can discuss it :open_hands:.

Reply:
(I just added a new unit test with the current changes and it worked ok, the snapshot showed the page updated. So it doesnt seem to be a blocker for now, but wonder if in the future we can find some race conditions because of this)

Reply:
I understand your concern. There will be changes in the way tests are written currently. We can discuss this once we are done working on the backbone of the visual testing mechanism. Once we have the setup complete, we can look for ways to streamline the tests.